Sources/SwiftDocC/Infrastructure/Bundle Assets/DataAssetManager.swift

```diff
@@ -354,9 +354,4 @@ public struct AssetReference: Hashable, Codable {
         self.assetName = assetName
         self.bundleIdentifier = bundleIdentifier
     }
-    
-    public func hash(into hasher: inout Hasher) {
```

Comment:
Does Swift also synthesize a default initializer? For example here in `AssetReference` could we also remove this code?

```
   /// Creates a reference from a given asset name and the bundle it is apart of.
    public init(assetName: String, bundleIdentifier: String) {
        self.assetName = assetName
        self.bundleIdentifier = bundleIdentifier
    }
```

Maybe the `public` keyword is the problem - I'm guessing the synthesized initializer would not be public, and that there's no way to mark the generated initializer as public.

Reply:
Correct. If a structure doesn't declare an explicit initializer, Swift will synthesize a default member-wise initializer. This initializer is _internal_, so callers from other modules can't access it. 

[SE-0018] proposed one solution for this in the early days of Swift but that proposal received a lot of feedback and was "returned for revision". AFAICT it wasn't revised and didn't get another review.